Adds cmake target and gtests for `health_monitor` code (#29)
kestrel-labs/superspreader: arduino/CMakeLists.txt

cmake_minimum_required(VERSION 3.5)

project(
  "superspreader"
  VERSION 1.0.0
  DESCRIPTION "Code for the game: \"Super Spreader\""
)

#############################################################################
# Super Spreader Libraries
#############################################################################

> include(GNUInstallDirs)

Reviewer: ```suggestion
```

cmake_minimum_required(VERSION 3.5)

project(
  "superspreader"
  VERSION 1.0.0
  DESCRIPTION "Code for the game: \"Super Spreader\""
)

#############################################################################
# Super Spreader Libraries
#############################################################################

# Default to Release build type
if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE Release)
endif()

# Use O3 on Release mode
set(CMAKE_CXX_FLAGS_RELEASE "-O3")

> set(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR})

Reviewer: ```suggestion
```

cmake_minimum_required(VERSION 3.5)

project(
  "superspreader"
  VERSION 1.0.0
  DESCRIPTION "Code for the game: \"Super Spreader\""
)

#############################################################################
# Super Spreader Libraries
#############################################################################

include(GNUInstallDirs)

> set(CMAKE_CXX_STANDARD 14)
> set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -O3")

Reviewer: I don't think you want to do either of these things. Instead, it is more idiomatic in cmake to set the build type for the optimization level. To make sure Release build type is using `O3` (I think it uses `O2` by default) you can do something like this:

```suggestion
# Default to Release build type
if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE Release)
endif()

# Use O3 on Release mode
set(CMAKE_CXX_FLAGS_RELEASE "-O3")
```
